parquet-column/src/main/java/org/apache/parquet/column/values/dictionary/IntList.java

```diff
@@ -31,7 +31,13 @@
  */
 public class IntList {
 
-  private static final int SLAB_SIZE = 64 * 1024;
+  public static final int MAX_SLAB_SIZE = 64 * 1024;
+  public static final int INITIAL_SLAB_SIZE = 4 * 1024;
+
+  //Double slab size till we reach the max slab size. At that point we just add slabs of size
+  //MAX_SLAB_SIZE. This ensures we don't allocate very large slabs from the start if we don't have
+  //too much data.
+  private int currentSlabSize = INITIAL_SLAB_SIZE / 2;
```

Comment:
That works for me. You could also rename `currentSlabSize` to `nextSlabSize` so it's more clear. Checks should all be in terms of some slab's `int[]#length` anyway.